lib/src/shared/main/java/com/couchbase/lite/AbstractReplicator.java

```diff
@@ -392,52 +395,56 @@ public void stop() {
     public Status getStatus() { return status.copy(); }
 
     @NonNull
-    public Set<String> getPendingDocumentIds() throws CouchbaseLiteException {
+    Set<String> getPendingDocumentIds() throws CouchbaseLiteException {
         if (config.getReplicatorType().equals(ReplicatorConfiguration.ReplicatorType.PULL)) {
             throw new CouchbaseLiteException(
                 "Pending Document IDs are not supported on pull-only replicators.",
-                Log.LOGGING_DOMAINS_TO_C4.get(DOMAIN),
+                CBLError.Domain.CBLITE,
                 CBLError.Code.UNSUPPORTED);
         }
 
         final C4Replicator repl;
-        synchronized (lock) { repl = c4repl; }
+        synchronized (lock) { repl = c4Replicator; }
 
-        // !!! Need to return a result here: will require recreating the replicator.
+        // ??? Not yet implemented
         if (repl == null) {
             Log.i(DOMAIN, "%s: Call to getPendingDocumentIds with unstarted replicator", this);
             return Collections.emptySet();
         }
 
         final Set<String> pending;
-        try { pending = c4repl.getPendingDocIDs(); }
+        try { pending = repl.getPendingDocIDs(); }
         catch (LiteCoreException e) { throw CBLStatus.convertException(e, "Failed fetching pending documentIds"); }
 
-        return (pending.size() <= 0) ? Collections.emptySet() : Collections.unmodifiableSet(pending);
+        // ??? C4 call returns null if the native replicator is gone: Not yet implemented
+        return (pending == null) ? Collections.emptySet() : Collections.unmodifiableSet(pending);
     }
 
-    @NonNull
-    public boolean isDocumentPending(@NonNull String docId) throws CouchbaseLiteException {
+    boolean isDocumentPending(@NonNull String docId) throws CouchbaseLiteException {
         Preconditions.checkArgNotNull(docId, "document ID");
 
         if (config.getReplicatorType().equals(ReplicatorConfiguration.ReplicatorType.PULL)) {
             throw new CouchbaseLiteException(
                 "Pending Document IDs are not supported on pull-only replicators.",
-                Log.LOGGING_DOMAINS_TO_C4.get(DOMAIN),
+                CBLError.Domain.CBLITE,
                 CBLError.Code.UNSUPPORTED);
         }
 
         final C4Replicator repl;
-        synchronized (lock) { repl = c4repl; }
+        synchronized (lock) { repl = c4Replicator; }
 
-        // !!! Need to return a result here: will require recreating the replicator.
+        // ??? Not yet implemented
         if (repl == null) {
             Log.i(DOMAIN, "%s: Call to isDocumentPending with unstarted replicator", this);
             return false;
         }
 
-        try { return c4repl.isDocumentPending(docId); }
+        final Boolean pending;
+        try { pending = repl.isDocumentPending(docId); }
```

Comment:
same, may need to lock

Reply:
No.  No lock needed.  Please see the c4Replicator code.